cogfiles/reaction_roles.py

```diff
@@ -0,0 +1,139 @@
+
+        reactionrole = ReactionRole(channel, message, role, emoji)
+
+        # If there already exists a reaction role for this role, remove it and replace it with the new one
+        old_reactionrole = next((rr for rr in self.reactionroles[interaction.guild.id] if rr.role == reactionrole.role), None)
+        self.reactionroles[interaction.guild.id].discard(old_reactionrole)
+        self.reactionroles[interaction.guild.id].add(reactionrole)
+        write_json(interaction.guild.id, "reaction_roles", value=[reactionrole.to_json() for reactionrole in self.reactionroles[interaction.guild.id]])
+
+        embed = RandomColorEmbed(title="Reaction Role", description=f"React to this message with {emoji} to get the {role.mention} role: {message.jump_url}")
+        if old_reactionrole is not None:
+            embed.description += f"\nThis command replaced an existing reaction role for {role.mention} at {old_reactionrole.message.jump_url}"
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+
+    @app_commands.command()
+    @app_commands.checks.has_permissions(manage_guild=True)
+    async def reactionroleremove(self, interaction: discord.Interaction, role: discord.Role):
+        "Removes a reaction role message."
+        # If there already exists a reaction role for this role, remove it and replace it with the new one
+        reactionrole = next((rr for rr in self.reactionroles[interaction.guild.id] if rr.role == role), None)
+        if reactionrole is None:
+            return await interaction.response.send_message("Could not find a reaction role for that role.", ephemeral=True)
+        
+        await reactionrole.message.remove_reaction(reactionrole.emoji, self.bot.user)
+        self.reactionroles[interaction.guild.id].discard(reactionrole)
+        write_json(interaction.guild.id, "reaction_roles", value=[reactionrole.to_json() for reactionrole in self.reactionroles[interaction.guild.id]])
+
+        embed = RandomColorEmbed(title="Reaction Role Removed", description=f"\nThe reaction role for {role.mention} at {reactionrole.message.jump_url} has been removed.")
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+    
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent):
+        """Called when a message has a reaction added. 
+        This is called regardless of the state of the internal message cache, for example with old messages."""
+        guild = self.bot.get_guild(payload.guild_id)
+        member = guild.get_member(payload.user_id)
+        role = self.get_role(payload)
+
+        if role is None:
+            return
+
+        await member.add_roles(role)
+
+        if member.dm_channel is None:
+            await member.create_dm()
+        await member.dm_channel.send(f"You now have the {role.name} role in {guild.name}.")
```

Comment:
Up to you, but one thing we could do is enclose {role.name} (and maybe {guild.name}) in either bold or code brackets to highlight them in the text.